### PR TITLE
chore(deps): update to eslint-config-standard ^16.0.0

### PR DESCRIPTION
**Pull request by charles-allen**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[ ] Bug fix
[ ] New feature
[X] Other: Bump deps to the latest eslint-config-standard

**What changes did you make? (Give an overview)**

1. updated `eslint-config-standard` to ^16.0.0
1. aligned peer dep versions to `eslint-config-standard` 16.0.0
1. ~~aligned peer & dev deps of `@typescript-eslint/eslint-plugin`~~  _eslint-plugin is unchanged_
1. updated tests to reflect latest rules:
    - "peer deps should use ^ operator" standard/eslint-config-standard@ffc3c7594f9d3432b1759ca155bfea3808dcaa79  
      => test now strips `^` & `>=` before comparison
    - "ignore jsx nodes for 'indent' rules" standard/eslint-config-standard@77f16926334c88724efca1b25ae59ecb02f07edf  
      => updated `ignoredNodes` in expectation
    - "require indentation for values of ternary expressions" standard/eslint-config-standard@efc8318a96e89caac3583ed6ea901e389f5afa3a
      => added `offsetTernaryExpressions` to expectation
    - "make default option explicit" standard/eslint-config-standard@65fb1dc2f72c2128789c237d0ec271279997ffb1
      => updated `no-unused-vars` in expectation
1. removed `eslint-plugin-standard` & removed from docs standard/eslint-config-standard@0f4d3f2fb91e55d13548736f4bf4b639e2ee1a57
1. tried to follow commit message rules :)

**Which issue (if any) does this pull request address?**

_none_

**Is there anything you'd like reviewers to focus on?**

I hope I did it correctly!

### CODE REVIEW

**Comment by hugomrdias** [comment]
🙏  Thank you

**Comment by charles-allen** [comment]
@mightyiam - Can we merge this? :)
Do you need anything else changing?

File: package.json [hunk 1]
```diff
@@ -60,31 +60,29 @@
...
+    "typescript": "^3.9",
+    "@typescript-eslint/eslint-plugin": "^4.12.0"
   },
   "devDependencies": {
     "@commitlint/cli": "11.0.0",
     "@commitlint/travis-cli": "11.0.0",
     "@types/eslint": "7.2.6",
     "@types/node": "14.14.20",
-    "@typescript-eslint/eslint-plugin": "4.0.1",
+    "@typescript-eslint/eslint-plugin": "4.12.0",
```
Comment (mightyiam) [inline package.json:78]: Is this related?

File: src/index.test.ts [hunk 2]
```diff
@@ -173,7 +174,7 @@ test('Dependencies range types', async (t) => {
     t.true(range.startsWith(specifier), `Regular dependency ${name} starts with \`${specifier}\`.`)
   }
   for (const [name, range] of Object.entries(ourPeerDeps)) {
-    const specifier = '>='
+    const specifier = '^'
```
Comment (mightyiam) [inline src/index.test.ts:177]: Why?

File: package.json [hunk 3]
```diff
@@ -60,31 +60,29 @@
...
-    "eslint-plugin-promise": ">=4.2.1",
-    "eslint-plugin-standard": ">=4.0.0",
-    "typescript": ">=3.9",
-    "@typescript-eslint/eslint-plugin": ">=4.0.1"
+    "eslint": "^7.12.1",
+    "eslint-plugin-import": "^2.22.1",
+    "eslint-plugin-node": "^11.1.0",
+    "eslint-plugin-promise": "^4.2.1",
+    "typescript": "^3.9",
+    "@typescript-eslint/eslint-plugin": "^4.12.0"
```
[the remark this answers is not in the document]
Comment (charles-allen) [reply package.json:71]: Yes, see change in standard:
"peer deps should use ^ operator" standard/eslint-config-standard@ffc3c75

File: src/index.test.ts (hunk 2 above)
Comment (charles-allen) [reply src/index.test.ts:177]: eslint-config-standard moved peer deps to '^' here: standard/eslint-config-standard@ffc3c75
The tests here require me to use the same versions of peer deps.

File: package.json (hunk 1 above)
Comment (charles-allen) [reply package.json:78]: I think that was from renovate; I just aligned dev & peer deps.
Should I revert both to 4.0.1?

Comment (charles-allen) [reply package.json]: I've reverted this change.

File: package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (charles-allen) [reply package.json]: I've gone ahead and switched to option 2:
- peer deps use `>=` again
- test `Own peerDependencies include those of eslint-config-standard` now strips `^` from standard deps & strips `>=` from `ourRange`, then compares the version number _(note: it does also catch if either symbol is missing or different from expected since the array index will fail)_

This also meant that I could revert my changes to the other tests. The diff is now smaller & more focused.

(this update is all together in 1 commit so if you choose option 1 I can just revert that commit => I'm ready for either option!)

File: src/index.test.ts (hunk 2 above)
Comment (charles-allen) [reply src/index.test.ts]: This change is now reverted by switching to option 2 above

File: src/index.test.ts [hunk 4]
```diff
@@ -88,7 +88,8 @@ test('export', (t): void => {
             ImportDeclaration: 1,
             flatTernaryExpressions: false,
             ignoreComments: false,
-            ignoredNodes: ['TemplateLiteral *']
+            ignoredNodes: ['TemplateLiteral *', 'JSXElement', 'JSXElement > *', 'JSXAttribute', 'JSXIdentifier', 'JSXNamespacedName', 'JSXMemberExpression', 'JSXSpreadAttribute', 'JSXExpressionContainer', 'JSXOpeningElement', 'JSXClosingElement', 'JSXFragment', 'JSXOpeningFragment', 'JSXClosingFragment', 'JSXText', 'JSXEmptyExpression', 'JSXSpreadChild'],
+            offsetTernaryExpressions: true
```
Comment (mightyiam) [inline src/index.test.ts:92]: https://github.com/standard/eslint-config-standard/blob/master/eslintrc.json

File: src/index.test.ts [hunk 5]
```diff
@@ -127,7 +128,7 @@ test('export', (t): void => {
           '@typescript-eslint/no-throw-literal': 'error',
           '@typescript-eslint/no-unnecessary-type-assertion': 'error',
           '@typescript-eslint/no-unnecessary-boolean-literal-compare': 'error',
-          '@typescript-eslint/no-unused-vars': ['error', { vars: 'all', args: 'none', ignoreRestSiblings: true }],
+          '@typescript-eslint/no-unused-vars': ['error', { args: 'none', caughtErrors: 'none', ignoreRestSiblings: true, vars: 'all' }],
```
Comment (mightyiam) [inline src/index.test.ts:131]: https://github.com/standard/eslint-config-standard/blob/master/eslintrc.json

File: src/index.test.ts [hunk 6]
```diff
@@ -189,7 +190,9 @@ test('Own peerDependencies include those of eslint-config-standard', async (t) =
       // https://github.com/microsoft/TypeScript/pull/12253
       const name = _name as keyof typeof standardPkg.peerDependencies
       const ourRange = ourPeerDeps[name]
-      t.is(ourRange, standardRange)
+      const ourMinimum = ourRange.split('>=')[1]
+      const standardMinimum = standardRange.split('^')[1]
```
Comment (mightyiam) [inline src/index.test.ts:194]: A naming concern here.
The term minimum can apply to `>=` but not to `^`, because `^` has a maximum, as well. So it's not just a minimum, but a range.

Comment (charles-allen) [reply src/index.test.ts]: Updated as:
```
test('Own peerDependencies include those of eslint-config-standard', async (t) => {
  const { ourPeerDeps } = await getOurDeps()
  Object
    .entries(standardPkg.peerDependencies)
    .forEach(([_name, standardDep]) => {
      // https://github.com/microsoft/TypeScript/pull/12253
      const name = _name as keyof typeof standardPkg.peerDependencies
      const ourDep = ourPeerDeps[name]
      const ourRange = ourDep.split('>=')[1]
      const standardRange = standardDep.split('^')[1]
      t.is(ourRange, standardRange)
    })
})
```

**Comment by mightyiam** [comment]
Resolve conflicts, please.

**Comment by charles-allen** [comment]
> Resolve conflicts, please.

> This branch has no conflicts with the base branch

Errr... what conflicts?!

File: src/index.test.ts (hunk 4 above)
Comment (charles-allen) [reply src/index.test.ts:92]: Not sure what this comment wants. I assumed it was just to log where the updated values came from. Looks aligned to me:
```
// eslint-config-standard
"ignoredNodes": ["TemplateLiteral *", "JSXElement", "JSXElement > *", "JSXAttribute", "JSXIdentifier", "JSXNamespacedName", "JSXMemberExpression", "JSXSpreadAttribute", "JSXExpressionContainer", "JSXOpeningElement", "JSXClosingElement", "JSXFragment", "JSXOpeningFragment", "JSXClosingFragment", "JSXText", "JSXEmptyExpression", "JSXSpreadChild"],

// eslint-config-standard-with-typescript (test)
ignoredNodes: ['TemplateLiteral *', 'JSXElement', 'JSXElement > *', 'JSXAttribute', 'JSXIdentifier', 'JSXNamespacedName', 'JSXMemberExpression', 'JSXSpreadAttribute', 'JSXExpressionContainer', 'JSXOpeningElement', 'JSXClosingElement', 'JSXFragment', 'JSXOpeningFragment', 'JSXClosingFragment', 'JSXText', 'JSXEmptyExpression', 'JSXSpreadChild'],
```

File: src/index.test.ts (hunk 5 above)
Comment (charles-allen) [reply src/index.test.ts:131]: Same here; assume this is just a note; looks aligned:
```
// eslint-config-standard
"no-unused-vars": ["error", { "args": "none", "caughtErrors": "none", "ignoreRestSiblings": true, "vars": "all" }],

// eslint-config-standard-with-typescript (test)
'@typescript-eslint/no-unused-vars': ['error', { args: 'none', caughtErrors: 'none', ignoreRestSiblings: true, vars: 'all' }],
```

**Comment by charles-allen** [comment]
Thank you @mightyiam :)
